graphics/tinygl/clip.cpp

```diff
@@ -226,13 +226,12 @@ float(*clip_proc[6])(Vector4 *, Vector4 *, Vector4 *) =  {
 
 static inline void updateTmp(GLContext *c, GLVertex *q, GLVertex *p0, GLVertex *p1, float t) {
 	if (c->current_shade_model == TGL_SMOOTH) {
-		float a = q->color.W;
 		q->color = p0->color + (p1->color - p0->color) * t;
-		q->color.W = a;
```

Comment:
The fix looks good to me. There's no reason to do special handling for the alpha component when interpolating.

Out of curiosity, I checked the disassembly for that function on x86_64 with GCC6.1 / O2. No Vector4 temporaries are created at all. Probably thanks to "return value optimization". The generated code is quite optimized, but not as much as I would have hoped. SIMD instructions are not used to perform the operations on all the vector components at once.

BTW, the assignment below can be done as "q->color = p0->color", now.